tests/subsys/fs/fs_api/src/test_fs_dir_file.c

```diff
@@ -119,20 +213,56 @@ void test_unmount(void)
 
 	TC_PRINT("\nunmount file system that has never been mounted:\n");
 	ret = fs_unmount(&test_fs_mnt_unsupported_fs);
-	zassert_equal(ret, -EINVAL, "Unmount a never mounted fs");
+	zassert_not_equal(ret, 0, "Unmount a never mounted fs");
 
-	TC_PRINT("\nunmount file system multiple times:\n");
 	ret = fs_unmount(&test_fs_mnt_1);
 	zassert_true(ret >= 0, "Fail to unmount fs");
 
+	TC_PRINT("\nunmount file system multiple times:\n");
 	test_fs_mnt_1.fs = &temp_fs;
 	ret = fs_unmount(&test_fs_mnt_1);
-	zassert_equal(ret, -EINVAL, "Unmount a unmounted fs");
+	zassert_not_equal(ret, 0, "Unmount a unmounted fs");
+
+	TC_PRINT("unmount an unmountable file system\n");
+	ret = fs_unmount(&test_fs_mnt_no_op);
+	zassert_not_equal(ret, 0, "Unmount a unmountable fs");
+	/* TEST_FS_2 is registered in test_mount(), unregister it here*/
+	fs_unregister(TEST_FS_2, &null_fs);
 }
 
 /**
  * @brief Test statvfs interface of filesystem
  *
+ * @details
+ * Test Objective
+ * - Test fs_statvfs() interface in file system
+ *
+ * Testing techniques:
+ * - Interface testing
+ * - function and block box testing
+ * - dynamic analysis and testing
```

Comment:
I do not understand the claim.

Reply:
These testing techniques are described in IEC-61508, in test report for every test case, the "testing techniques" need to be listed

Reply:
What I meant is that, as far as I understand, the "dynamic analysis" is form of testing itself so addition of "testing" does change it's nature; additionally the subsection is called "Testing Techniques" so listing "testing" as a method seems redundant.

Reply:
> 
> These testing techniques are described in IEC-61508, in test report for every test case, the "testing techniques" need to be listed

@mengxianglinx OK, so this is part of bigger effort. I am not familiar with the IEC-61508, and my brief scroll through the documents did not make me an expert, but shouldn't the entire form/structure of comments be discussed and defined prior to starting with the effort?
I guess that this is going to be implemented in all other tests.
Have other forms been considered and the effect of bloating the code taken into account?
Has there been a template provided, so that it could be used instead of copy-paste from other test cases?

Reply:
Thank you @de-nordic ! Our team is trying to simplify document form/structure to avoid bloating the code. Today we have an online meeting about Function Safety [test work group](https://github.com/zephyrproject-rtos/zephyr/wiki/Zephyr-Committee-and-Working-Group-Meetings#testing-working-group) to discuss about these comment formats. We are appreciated if you can attend this meeting.